client/components/domains/register-domain-step/index.jsx

```diff
@@ -263,9 +263,30 @@ class RegisterDomainStep extends Component {
 		this._isMounted = false;
 	}
 
+	// In the launch flow, the initial query could sometimes be missing if the user had
+	// created a site by skipping the domain step. In these cases, fire the initial search
+	// with the subdomain name.
+	getInitialQueryInLaunchFlow() {
+		if ( ! this.props.isInLaunchFlow ) {
+			return;
+		}
+
+		if (
+			typeof this.props.selectedSite !== 'object' ||
+			typeof this.props.selectedSite.domain !== 'string'
+		) {
+			return;
+		}
+
+		const wpcomSubdomainWithRandomNumberSuffix = /^(.+?)([0-9]{5,})\.wordpress\.com$/i;
```

Comment:
Don't we have convenience functions for this elsewhere in domain libs? 

I'd expect such function strip out also free .blog subdomains, and other such domains in case we add more.

Reply:
I refactored and removed the match to ".wordpress.com" since it's not actually needed - the current regex should work against any subdomain.